tests/build_provider_packages_dependencies.py

```diff
@@ -0,0 +1,269 @@
+        Retrieves import name from the "from" import.
+        :param node: ImportFrom name
+        :return: import name
+        """
+        import_names: List[str] = []
+        for alias in node.names:
+            name = alias.name
+            fullname = '%s.%s' % (node.module, name) if node.module else name
+            import_names.append(fullname)
+        return import_names
+
+    # noinspection PyPep8Naming
+    def visit_Import(self, node: Import):  # pylint: disable=invalid-name
+        for alias in node.names:
+            self.process_import(alias.name)
+
+    # noinspection PyPep8Naming
+    def visit_ImportFrom(self, node: ImportFrom):  # pylint: disable=invalid-name
+        if node.module == '__future__':
+            return
+        for fullname in self.get_import_name_from_import_from(node):
+            self.process_import(fullname)
+
+
+def get_imports_from_file(file_name: str) -> List[str]:
+    """
+    Retrieves imports from file.
+    :param file_name: name of the file
+    :return: list of import names
+    """
+    with open(file_name) as f:
+        root = parse(f.read(), file_name)
+    visitor = ImportFinder(file_name)
+    visitor.visit(root)
+    return visitor.imports
+
+
+def check_if_different_provider_used(file_name: str):
+    file_provider = get_provider_from_file_name(file_name)
+    if not file_provider:
+        return
+    imports = get_imports_from_file(file_name)
+    for import_name in imports:
+        import_provider = get_provider_from_import(import_name)
+        if import_provider and file_provider != import_provider:
+            dependencies[file_provider].append(import_provider)
+
+
+def parse_arguments():
+    import argparse
+    parser = argparse.ArgumentParser(
+        description='Checks if dependencies between packages are handled correctly.')
+    parser.add_argument("-f", "--provider-dependencies-file",
+                        help="Stores dependencies between providers in the file")
+    parser.add_argument("-d", "--documentation-file",
+                        help="Updates package documentation in the file specified (.rst)")
+    parser.add_argument('files', nargs='*')
+    args = parser.parse_args()
+
+    if len(args.files) < 1:
+        parser.print_usage()
+        print()
+        sys.exit(2)
+    return args.files, args.provider_dependencies_file, args.documentation_file
+
+
+PREFIX = "    "
+
+HEADER = """
+========================== ===========================
+Package                    Extras
+========================== ===========================
+"""
+FOOTER = """========================== ===========================
+
+"""
+
+
+def insert_documentation(deps_dict: Dict[str, List[str]], res: List[str]):
+    res += HEADER.splitlines(keepends=True)
+    for package, deps in deps_dict.items():
+        deps_str = ",".join(deps)
+        res.append(f"{package:27}{deps_str}\n")
+    res += FOOTER.splitlines(keepends=True)
+
+
+if __name__ == '__main__':
+    print()
```

Comment:
Used it for debugging? We can remove it

Reply:
No it's actually used to generate list of dependencies and write them to documentation in pre-commit :).